frontend/src/components/SignUp.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import { useSelector, useDispatch } from 'react-redux'
+import styled from 'styled-components'
+
+import { StyledButton } from '../components/StyledBtn'
+import { API_URL } from '../reusables/urls'
+import user from '../reducers/user'
+
+const Wrapper = styled.div`
+  border: 1px solid #9099A5;
+  border-radius: 2px;
+  box-shadow: 7px 7px #CAEBF2, 7px 7px 0px 1px #9099A5;
+  background-color: white;
+  margin: 30px;
+  width: 500px;
+  height: 250px;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+`
+
+const Form = styled.form`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+`
+
+const Input = styled.input`
+  margin: 7px;
+`
+
+export const SignUp = () => {
+  const [ username, setUsername ] = useState('')
+  const [ password, setPassword ] = useState('')
+
+  const history = useHistory()
+  const dispatch = useDispatch()
+  const errorMessage = useSelector(store => store.user.errors)
+  
+
+  const onFormSubmit = (e) => {
+    e.preventDefault()
+    fetch(API_URL('signup'), {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        username,
+        password
+      })
+    })
+      .then(response => response.json())
+      .then(data => {
+        if (data.success) {
+          history.push('/login')
```

Comment:
Why? Your API already provided you with all credentials here so why not log your user behind the scenes and redirect straight to the content?